### PR TITLE
Interleaving url hinting.

### PR DESCRIPTION
**Pull request by amosbird**
Implements #595

### CODE REVIEW

File: lib/select_wm.lua [hunk 1]
```diff
@@ -60,6 +60,34 @@ local label_styles = {
         return function (size) return charset("0123456789", size) end
     end,
 
+    -- Interleave style
+    interleave = function ()
+        return function (size)
+            local sub, concat = string.sub, table.concat
+            local left, right = "dsrewvcxg", "kluionmhb" -- equal length
```
Comment (aidanholm) [inline lib/select_wm.lua]: I'd prefer that the left and right strings be passed as parameters. People might want to alternate between 1--5 and 6--0, for example. Is it possible to make this work with the parameters not equal length?

Comment (aidanholm) [reply lib/select_wm.lua]: Also, does this function only return sequences that go LRLR...? It should also return sequences that start with a right-hand key (RLRL...)

Comment (amosbird) [reply lib/select_wm.lua]: > Is it possible to make this work with the parameters not equal length?

It has to be equal length for now. I'll add a parameter for that with a length check then :)

> Also, does this function only return sequences that go LRLR...? It should also return sequences that start with a right-hand key (RLRL...)

it generates both LR and RL patterns

File: lib/select_wm.lua [hunk 2]
```diff
@@ -60,6 +60,35 @@ local label_styles = {
...
+        return function (size)
+            local sub, concat = string.sub, table.concat
+            local base, digits, labels = #left, {}, {}
+            digits[0], digits[1] = {}, {}
+            for i = 1, base do
+                rawset(digits[1], i, sub(left, i, i))
+                rawset(digits[0], i, sub(right, i, i))
+            end
+            local maxlen = max_hint_len(ceil(size/2), base)
+            local zeroseq = string.rep("fj", maxlen)
```
Comment (aidanholm) [inline lib/select_wm.lua]: Left-over hardcoded value here: should be `left[1] .. right[1]` I guess?

Comment (amosbird) [reply lib/select_wm.lua]: hmm, right. I'll pop out one from left and right to use as zerochar :)

**Pull request merged**